### PR TITLE
[Improvement] Parallelize github workflow to speed up CI

### PR DESCRIPTION
**Pull request by kaijchen**
### What changes were proposed in this pull request?

Parallelize github workflow to speed up CI.

### Why are the changes needed?

Reduced wall-clock time of CI from ~1 hour to ~25 minutes.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

https://github.com/kaijchen/Firestorm/actions/runs/2362806351

### CODE REVIEW

**Comment by frankliee** [comment]
It seems that the CI pipeline is not launched automatically after PR is submitted.
It hash to be run by hand.

**Comment by jerqi** [comment]
Great work!

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -39,27 +47,11 @@ jobs:
...
+        name: test-results-of-rss-${{ matrix.profile }}
         path: "**/target/surefire-reports/*.txt"
-    - name: Build with Maven with Profile mr
-      run: mvn -B package --file pom.xml -Pmr
-    - name: Upload mr test results to report
-      if: failure()
-      uses: actions/upload-artifact@v2
-      with:
-        name: test-results-of-rss-mr
-        path: "**/target/surefire-reports/*.txt"
```
Comment (jerqi) [inline .github/workflows/maven.yml:65]: You'd better have a new line.

**Review by jerqi — APPROVED** [review]

**Comment by jerqi** [comment]
Thanks for your contribution.

**Comment by kaijchen** [comment]
Thanks @jerqi and @frankliee for the review.

**Pull request merged**